ibis/expr/operations.py

```diff
@@ -1612,10 +1640,20 @@ class AsOfJoin(Join):
 
     def __init__(self, left, right, predicates, by_predicates):
         _validate_join_tables(left, right)
+
+        if not isinstance(predicates, (tuple, list)):
```

Comment:
I feel like there's a `util.to_list` somewhere in ibis. Does that not work here because of the `tuple`?

Reply:
Yep, `util.promote_list` only works with list